### PR TITLE
Obtains the actual load data of a node from the custom metrics API.

### PR DESCRIPTION
**Pull request by wangyang0616**
Related: [#3175](https://github.com/volcano-sh/volcano/issues/3175)

1. Usage monitoring indicators support obtaining monitoring indicators from "Prometheus" and "Eleasticsearch", and add support for "Custom Metrics API"
2. Optimize usage configuration usage information. The optimized configuration information is as follows:
```
 - name: usage  # usage based scheduling plugin
        enablePredicate: false  # If the value is false, new pod scheduling is not disabled when the node load reaches the threshold. If the value is true or left blank, new pod scheduling is disabled.
        arguments:
          usage.weight: 5
          cpu.weight: 1
          memory.weight: 1
          thresholds:
            cpu: 80    # The actual CPU load of a node reaches 80%, and the node cannot schedule new pods.
            mem: 70    # The actual Memory load of a node reaches 70%, and the node cannot schedule new pods.
```
3. By default, Volcano queries monitoring indicator information every 30 seconds. The monitoring indicators are the average utilization of the CPU and Memory of each node in the last ten minutes.
4. If monitoring information cannot be obtained for more than 5 minutes, usage will not take effect. In the filtering phase, the usage plugin will not intercept nodes that have not obtained monitoring information. In the optimization phase, the usage docking point’s score is 0.

### CODE REVIEW

File: docs/design/usage-based-scheduling.md [hunk 1]
```diff
@@ -57,9 +62,9 @@ tiers:
       - name: nodeorder
       - name: binpack
 metrics:                               # metrics server related configuration
-  type: prometheus                     # Optional, The metrics source type, prometheus by default, support prometheus and elasticsearch
```
Comment (william-wang) [inline docs/design/usage-based-scheduling.md:60]: "prometheus_adapt" should be "prometheus_adaptor"

File: docs/design/usage-based-scheduling.md [hunk 2]
```diff
@@ -57,9 +62,9 @@ tiers:
       - name: nodeorder
       - name: binpack
 metrics:                               # metrics server related configuration
-  type: prometheus                     # Optional, The metrics source type, prometheus by default, support prometheus and elasticsearch
+  type: prometheus                     # Optional, The metrics source type, prometheus by default, support "prometheus", "prometheus_adapt" and "elasticsearch"
   address: http://192.168.0.10:9090    # Mandatory, The metrics source address
-  interval: 30s                        # Optional, The scheduler pull metrics from Prometheus with this interval, 5s by default
```
Comment (william-wang) [inline docs/design/usage-based-scheduling.md:62]: Is the default 30s too long?

File: docs/design/usage-based-scheduling.md [hunk 3]
```diff
@@ -92,31 +97,124 @@ Finally, there should a model to balance multiple factors with weight and calcul
...
+      - name: proportion
+      - name: nodeorder
+      - name: binpack
+metrics:                               # metrics server related configuration
+  type: prometheus                     # Optional, The metrics source type, prometheus by default, support "prometheus", "prometheus_adapt" and "elasticsearch"
+  address: http://192.168.0.10:9090    # Mandatory, The metrics source address
+  interval: 30s                        # Optional, The scheduler pull metrics from Prometheus with this interval, 30s by default
+  ```
+#### Custom Metrics Api
+Ensure that Prometheus Adapt is properly installed in the cluster and the custom metrics API is available.
```
Comment (william-wang) [inline docs/design/usage-based-scheduling.md:133]: Prometheus Adaptor?

File: docs/design/usage-based-scheduling.md [hunk 4]
```diff
@@ -92,31 +97,124 @@ Finally, there should a model to balance multiple factors with weight and calcul
...
+Set the user-defined indicator information. The rules to be added are as follows. For details, see [Metrics Discovery and Presentation Configuration](https://github.com/kubernetes-sigs/prometheus-adapter/blob/master/docs/config.md#metrics-discovery-and-presentation-configuration)
+```azure
```
Comment (william-wang) [inline docs/design/usage-based-scheduling.md:135]: why it is "azure"?

File: docs/design/usage-based-scheduling.md [hunk 5]
```diff
@@ -92,31 +97,124 @@ Finally, there should a model to balance multiple factors with weight and calcul
...
+      - name: predicates
+      - name: proportion
+      - name: nodeorder
+      - name: binpack
+metrics:                               # metrics server related configuration
+  type: prometheus                     # Optional, The metrics source type, prometheus by default, support "prometheus", "prometheus_adapt" and "elasticsearch"
+  address: http://192.168.0.10:9090    # Mandatory, The metrics source address
+  interval: 30s                        # Optional, The scheduler pull metrics from Prometheus with this interval, 30s by default
+  ```
+#### Custom Metrics Api
```
Comment (william-wang) [inline docs/design/usage-based-scheduling.md:132]: please add notes that prometheus adaptor is recommended.

File: docs/design/usage-based-scheduling.md (hunk 2 above)
Comment (wangyang0616) [reply docs/design/usage-based-scheduling.md:62]: 1. The Prometheus does not collect data frequently. Generally, the data collection frequency ranges from 10s to 60s. The frequency of obtaining monitoring data from the Prometheus does not need to be too frequent.
2. In a large-scale cluster (1000+ nodes) scenario, the Custom Metrics API is used to frequently obtain monitoring information, which causes great pressure on kube-apiserver.
3. Load awareness scheduling does not have strict requirements on real-time resources. Therefore, the indicator collection period can be prolonged.

File: docs/design/usage-based-scheduling.md (hunk 4 above)
Comment (wangyang0616) [reply docs/design/usage-based-scheduling.md:135]: The format is automatically added during editing in the local IDE and has been deleted.

File: docs/design/usage-based-scheduling.md (hunk 5 above)
Comment (wangyang0616) [reply docs/design/usage-based-scheduling.md]: done

File: docs/design/usage-based-scheduling.md (hunk 1 above)
Comment (wangyang0616) [reply docs/design/usage-based-scheduling.md:60]: done

**Review by william-wang — APPROVED** [review]
/lgtm

**Pull request merged**